daemon/SecretsImpl/secrets.cpp

```diff
@@ -1814,14 +1822,60 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
                 qCWarning(lcSailfishSecretsDaemon) << "GetStandaloneSecretRequest:" << request->requestId << "finished as pending!";
                 *completed = true;
             } else {
-                QByteArray secret = request->outParams.size()
-                        ? request->outParams.takeFirst().toByteArray()
-                        : QByteArray();
+                Secret secret = request->outParams.size()
+                        ? request->outParams.takeFirst().value<Secret>()
+                        : Secret();
                 if (request->isSecretsCryptoRequest) {
-                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<QByteArray>(secret));
+                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<Secret>(secret));
                 } else {
                     request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
-                                                                            << QVariant::fromValue<QByteArray>(secret));
+                                                                            << QVariant::fromValue<Secret>(secret));
+                }
+                *completed = true;
+            }
+            break;
+        }
+        case FindCollectionSecretsRequest: {
+            Result result = request->outParams.size()
+                    ? request->outParams.takeFirst().value<Result>()
+                    : Result(Result::UnknownError,
+                             QLatin1String("Unable to determine result of FindCollectionSecretsRequest request"));
+            if (result.code() == Result::Pending) {
+                // shouldn't happen!
+                qCWarning(lcSailfishSecretsDaemon) << "FindCollectionSecretsRequest:" << request->requestId << "finished as pending!";
+                *completed = true;
+            } else {
+                QVector<Secret::Identifier> identifiers = request->outParams.size()
+                        ? request->outParams.takeFirst().value<QVector<Secret::Identifier> >()
+                        : QVector<Secret::Identifier>();
+                if (request->isSecretsCryptoRequest) {
+                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<QVector<Secret::Identifier> >(identifiers));
```

Comment:
This style makes the code hard to read. Suggestion:
* use C++11 initializer lists instead
* also, the compiler can figure out the template argument on its own in this case, no need to explicitly write it.

```cpp
asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, { QVariant::fromValue(identifiers) });
```

Reply:
Why is the current code hard to read?

Reply:
It's a matter of opinion, I guess. No offence, I just think the initializer list is easier to understand.

Reply:
I agree with initializer list too. It is more readable.